### PR TITLE
fix(AttributeManager): Added a method to import attributes from anoth…

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…er AttributeManager and an index mapping.

### CODE REVIEW

File: tests/basic/test-attribute.cpp [hunk 1]
```diff
@@ -415,6 +415,38 @@ void test_copy_manager( geode::AttributeManager& manager )
...
+
+    geode::AttributeManager manager3;
+    for( const auto i : geode::LRange( nb_elements - 2 ) )
+    {
+        old2new[i] = nb_elements - 3 - i;
+    }
+    manager3.resize( nb_elements - 2 );
+    manager3.import( manager, old2new );
+    test_attribute_types( manager3 );
+    test_number_of_attributes( manager3, 8 );
```
Comment (panquez) [inline tests/basic/test-attribute.cpp:436]: Could you test all some values ?

File: include/geode/basic/attribute.h [hunk 2]
```diff
@@ -437,6 +453,30 @@ namespace geode
...
+            for( const auto i : Indices{ old2new } )
+            {
+                const auto new_index = old2new[i];
+                if( new_index != NO_ID )
+                {
+                    OPENGEODE_EXCEPTION( new_index < nb_elements,
+                        "[VariableAttribute::extract] The given mapping "
+                        "contains values that go beyond the given number of "
+                        "elements." );
+                    attribute->values_[new_index] = values_.at( i );
```
Comment (panquez) [inline include/geode/basic/attribute.h:474]: Maybe it is better to use .value(x) and .set_value(x, T) ?

**Review by panquez — APPROVED** [review]

File: include/geode/basic/attribute.h [hunk 3]
```diff
@@ -577,6 +617,32 @@ namespace geode
...
+        std::shared_ptr< AttributeBase > extract(
+            absl::Span< const index_t > old2new,
+            index_t nb_elements,
+            AttributeBase::AttributeKey ) const override
+        {
+            std::shared_ptr< VariableAttribute< bool > > attribute{
+                new VariableAttribute< bool >{
+                    static_cast< bool >( default_value_ ), this->properties() }
+            };
+            attribute->values_ = values_;
```
Comment (BotellaA) [inline include/geode/basic/attribute.h:629]:
```suggestion
```

**Review by BotellaA — CHANGES_REQUESTED** [review]

**Review by BotellaA — APPROVED** [review]

**Pull request merged**